subprojects/docs/src/docs/userguide/authoring-builds/software-product/composite_builds.adoc

```diff
@@ -101,13 +101,12 @@ However, this does not support all use cases and including plugin builds like th
 === Restrictions on included builds
 
 Most builds can be included into a composite, including other composite builds.
-However there are some restrictions.
+However, there are some restrictions.
 
 Every included build:
 
-* must not have a `rootProject.name` the same as another included build.
-* must not have a `rootProject.name` the same as a top-level project of the composite build.
-* must not have a `rootProject.name` the same as the composite build `rootProject.name`.
+* must not have a `rootProject.name` that is the same as a top-level project name of the composite build.
+* must not have a `rootProject.name` that is the same as the composite build `rootProject.name`.
```

Comment:
❓ Regarding this: You reference that this is tested:
https://github.com/gradle/gradle/blob/54825879d3213a77b317e2464c997b9798bd3fe4/subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/IncludedBuildValidationIntegrationTest.groovy#L101-L110

Though it seems to me that the test verifies that the restriction is not there anymore: In the test the root build (with root project name `buildA`) includes another build in directory `buildB` with root project name `buildA`. And that seems to work. So this restriction can be dropped as well.

I am not sure in this whole discussion though if the `rootProject.name` of the included build comes into play at all, or if the discussion is about the "name"/path of the included build: IIUC, this is determined by the directory name of the included build.